src/components/Reports/GoalCalculator/SavingStatus/SavingStatus.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useMemo, useState } from 'react';
+import { CircularProgress, Typography } from '@mui/material';
+import { DateTime } from 'luxon';
+import { useTranslation } from 'react-i18next';
+import { useLocale } from 'src/hooks/useLocale';
+import { formatRelativeTime } from 'src/lib/intlFormat';
+import { useGoalCalculator } from '../Shared/GoalCalculatorContext';
+import { getGoalLastUpdated } from './helpers';
+
+export const SavingStatus: React.FC = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { goalCalculationResult, isMutating } = useGoalCalculator();
+
+  const goalCalculation = goalCalculationResult.data?.goalCalculation;
+  const lastSavedAt = useMemo(
+    () => (goalCalculation ? getGoalLastUpdated(goalCalculation) : null),
+    [goalCalculation],
+  );
+
+  // Rerender periodically to update the saved at time
+  const [_tick, setTick] = useState(0);
+  useEffect(() => {
+    const interval = setInterval(() => setTick((tick) => tick + 1), 5000);
```

Comment:
[Must-fix] Memory leak potential: interval not cleared on unmount in strict mode. Evidence: setInterval without ref storage. Impact: Multiple intervals in development. Fix: Store interval ref. [Category: File]